lib/blocs/leaderboard/leaderboard/leaderboard_state.dart

```diff
@@ -0,0 +1,70 @@
+part of 'leaderboard_cubit.dart';
+
+abstract class LeaderboardState extends Equatable {
+  const LeaderboardState();
+
+  @override
+  List<Object> get props => [];
+}
+
+class LeaderboardInitial extends LeaderboardState {}
+
+abstract class DailyLeaderboardState extends Equatable {
+  const DailyLeaderboardState();
+
+  @override
+  List<Object> get props => [];
+}
+
+class DailyLeaderboardInitial extends LeaderboardState {}
+
+class DailyLeaderboardLoading extends LeaderboardState {
+  const DailyLeaderboardLoading();
+}
+
+class DailyLeaderboardSuccess extends LeaderboardState {
+  final int myRank;
+  final Iterable<DailyLeaderboardRow> rankList;
+  const DailyLeaderboardSuccess(this.myRank, this.rankList);
+
+  @override
+  List<Object> get props => [myRank, rankList];
+}
+
+class DailyLeaderboardFailure extends LeaderboardState {
+  final String statusMessage;
+  const DailyLeaderboardFailure(this.statusMessage);
+
+  @override
+  List<Object> get props => [statusMessage];
+}
```

Comment:
```suggestion
class LeaderboardInitial extends LeaderboardState {}

class LeaderboardLoading extends LeaderboardState {
  const LeaderboardLoading();
}

class DailyLeaderboardSuccess extends LeaderboardState {
  final int myRank;
  final Iterable<DailyLeaderboardRow> rankList;
  const DailyLeaderboardSuccess(this.myRank, this.rankList);

  @override
  List<Object> get props => [myRank, rankList];
}

class LeaderboardFailure extends LeaderboardState {
  final String statusMessage;
  const LeaderboardFailure(this.statusMessage);

  @override
  List<Object> get props => [statusMessage];
}
```
Initial, Loading and Failure states are same for both leaderboards. Just have one set of these